### PR TITLE
[TG-113][feat] Add scenario converter

### PR DESCRIPTION
**Pull request by lizavetaageenko**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/garage-it/SmartHouse-backend/pull/24" target="_blank">ReviewNinja

File: src/API/scenario-converter/scenario-converter.controller.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import ScenarioConverter from '../../scenarios/scenario.converter.js';
+import APIError from '../helpers/APIError';
+import httpStatus from 'http-status';
+
+function getConvertedScenario(req, res, next) {
+    if (!req.query.scenarioConfig) {
+        const err = new APIError('No scenario JSON received!', httpStatus.BAD_REQUEST);
```
Comment (artem713) [inline src/API/scenario-converter/scenario-converter.controller.js]: Could you please move string to constant

File: src/API/scenario-converter/scenario-converter.controller.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+        return next(err);
+    }
+
+    const scenarioConfig = JSON.parse(req.query.scenarioConfig);
+
+    if (!scenarioConfig.conditions || !scenarioConfig.actions || !scenarioConfig.logicalOperator) {
+        const err = new APIError('Incorrect scenario JSON received!', httpStatus.BAD_REQUEST);
```
Comment (artem713) [inline src/API/scenario-converter/scenario-converter.controller.js]: Could you please move string to constant

File: src/API/scenario-converter/scenario-converter.controller.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+function getConvertedScenario(req, res, next) {
+    if (!req.query.scenarioConfig) {
+        const err = new APIError('No scenario JSON received!', httpStatus.BAD_REQUEST);
+
+        return next(err);
+    }
+
+    const scenarioConfig = JSON.parse(req.query.scenarioConfig);
+
+    if (!scenarioConfig.conditions || !scenarioConfig.actions || !scenarioConfig.logicalOperator) {
```
Comment (artem713) [inline src/API/scenario-converter/scenario-converter.controller.js]: Move this checks to the separate function

**Comment by kosich** [comment]
⭐

**Pull request merged**